db/document.go

```diff
@@ -334,6 +335,7 @@ func unmarshalDocumentWithXattr(docid string, data []byte, xattrData []byte, cas
 	if xattrData == nil || len(xattrData) == 0 {
 		// If no xattr data, unmarshal as standard doc
 		doc, err = unmarshalDocument(docid, data)
+		doc.migrationRequired = true
```

Comment:
Does unmarshalDocumentWithXattr get called for SDK inserts (i.e. there's no xattrData, but there also isn't any _sync in the body).  In that case we wouldn't want to mark as migrationRequired.

Reply:
Yeah looks like it does... I'll add a check for `_sync` data in the body. With this being the case I'll move it out of `unmarshalDocumentWithXattr` as it seems neater to do it above with the extra check.